ibeckermayer/teleport-interview
# Docker-ized app setup

Relatively large PR to nail down some of the core app structure, manage self-signed SSL, and run a Docker-ized development and production basic app.

To test:
- See the updated `README.md` for instructions for setting up the dev/prod environments. If anything is confusing or not adequately explained, the `README.md` should be updated.
- When you navigate to the app in the browser, you should see the login form:
  - Invalid email string (i.e. no "@" character), or empty email or password should cause the html5 form validator to complain (screenshot 1)
  - Invalid credentials should cause a 401 error message to display in the console (screenshot 2)
  - Valid credentials (username and password both `admin@goteleport.com`) should take you to the dashboard route (screenshot 3)
  - (Behavior should be exactly the same for both dev and prod setups)

![Screen Shot 2021-01-05 at 12 17 06 PM](https://user-images.githubusercontent.com/13578537/103694934-97f00680-4f50-11eb-9b71-aa7114e78263.png)
![Screen Shot 2021-01-05 at 12 19 00 PM](https://user-images.githubusercontent.com/13578537/103694936-97f00680-4f50-11eb-867d-20114f320820.png)
![Screen Shot 2021-01-05 at 12 21 17 PM](https://user-images.githubusercontent.com/13578537/103694949-9cb4ba80-4f50-11eb-986d-343eb6fe14c5.png)
